tests/main/presenters/test_search_summary.py

```diff
@@ -266,6 +266,38 @@ def test_subcategory_filters_are_available(self):
         summary_markup = search_summary.markup()
         assert "category <em>Analytics</em>" in summary_markup
 
+    def test_each_filter(self):
+        """
+        Test each filter individually adds a single text string to the summary text
+        """
+        for lot in self._g9_lots_by_slug:
+            self.request_args['lot'] = lot
+            for filter_group in g9_filter_groups:
+                for f in filter_group['filters']:
+                    request_args = self.request_args.copy()
+                    request_args[f['name']] = f['value']
+                    search_summary = SearchSummary('9', request_args, g9_filter_groups, self._g9_lots_by_slug)
+                    summary_markup = search_summary.markup()
+                    assert summary_markup.count("<em>") == 3  # the keyword, the lot, and one filter
```

Comment:
I think I'm reluctant to try and get something better until after we've refactored the widget completely. The current design has been in place for a while, and this would catch the cases that we broke in/after G9 go-live.

We can't very well validate that the correct labels are going in without reproducing the entire logic from the code that processes the summary manifest. Again I'm reluctant to go there until we've refactored that (and potentially moved the content into the frameworks repo, depending if any such separate content continues to exist).